compact.go

```diff
@@ -439,6 +455,9 @@ func populateBlock(blocks []Block, indexw IndexWriter, chunkw ChunkWriter) (*Blo
 
 		meta.Stats.NumChunks += uint64(len(chks))
 		meta.Stats.NumSeries++
+		for _, chk := range chks {
+			meta.Stats.NumSamples += uint64(binary.BigEndian.Uint16(chk.Chunk.Bytes()))
```

Comment:
We should eventually make this a first class property of the chunk interface. Strictly speaking not all compression formats would need this. But enforcing it also has negligible cost. In doubt one could have the `Size()` function just iterate through all elements.